artists.php

```diff
@@ -0,0 +1,120 @@
+require_once 'src/DatabaseConnector.php';
+require_once 'src/Models/ArtistsModel.php';
+require_once 'src/Entities/Artist.php';
+require_once 'src/Entities/Album.php';
+require_once 'src/Models/AlbumsModel.php';
+
+
+$db = DatabaseConnector::connect();
+
+$artistsModel = new ArtistsModel($db);
+$artist = new Artist();
+$artists = $artistsModel->getAllArtists();
+$albumsModel = new AlbumsModel($db);
+
+?>
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Music Player</title>
+    <script src="https://cdn.tailwindcss.com"></script>
+    <script defer src="play.js"></script>
+</head>
+<body>
+<div class="h-screen w-full bg-blue-950 flex text-white">
+    <main class="group grow h-screen">
+        <section class="group-[.minimised]:h-[calc(100%-6rem)] h-3/4 p-12 overflow-auto">
+            <div class="flex justify-between">
+                <h2 class="text-4xl font-bold mb-6">Artists</h2>
+                <a href="index.php" class="align-top">
+                    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="size-4 inline">
+                        <path stroke-linecap="round" stroke-linejoin="round" d="M9 15 3 9m0 0 6-6M3 9h12a6 6 0 0 1 0 12h-3" />
+                    </svg>
+                    back
+                </a>
+            </div>
+            <div class="grid md:grid-cols-2 lg:grid-cols-3 gap-5 items-start">
+
+                <?php
+                    $artistLoop = '';
+                    foreach ($artists as $artist) {
+                    $artistLoop .= "
+                        <div class='rounded p-3 bg-cyan-950'>
+                        <h4 class='mb-3 text-2xl font-bold'>{$artist->getArtistName()}</h4>";
+
+                    $albums = $albumsModel->getAlbumsByArtistId($artist->getId());
+                    foreach ($albums as $album) {
+                        $artistLoop .= "
+                    <div class='mb-3 flex justify-between items-center'>
+                        <img class='size-14' src='{$album->getArtworkURL()}' />
+                        <div class='w-3/4 px-3'>
+                            <h4 class='font-bold text-lg'>{$album->getAlbumName()}</h4>
+                            <p class='text-sm'>{$album->getSongCount()} songs</p>
+                        </div>
+                        <a href='artist.php?id={$artist->getId()}' class='hover:text-slate-500 hover:cursor-pointer'>
+                            <svg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24' stroke-width='1.5' stroke='currentColor' class='size-6'>
+                                <path stroke-linecap='round' stroke-linejoin='round' d='m12.75 15 3-3m0 0-3-3m3 3h-7.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z' />
+                            </svg>
+                        </a>
+                    </div>"; }
+                $artistLoop .= "</div>";
+                }
+                    echo $artistLoop;
```

Comment:
This is the chunk here that really should go into a display service. You should give it a method that takes the array of artists and the album model as params - that way it has everything it needs to display the info